### PR TITLE
Docs: add some API links to stream operator docs

### PR DESCRIPTION
**Pull request by ennru**
## Purpose

Use a new feature in `@apidoc` which makes it possible to link to anchors within API docs.
Use the Javadoc 11 default of not creating HTML frames and use the new Paradox `link_style` setting.

## References

https://github.com/lightbend/sbt-paradox-apidoc/pull/81

## Background Context

We discussed if the signatures add value to the stream operator docs and agreed that links to the API docs might be more useful.

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM

Will be quite a bit of work to add to all but worth it for consistency. We'll just have to draw straws who gets to do it I guess.

**Comment by ennru** [comment]
> Will be quite a bit of work to add to all but worth it for consistency. We'll just have to draw straws who gets to do it I guess.

Agreed, that's why I built the support now so that it can be added with every updated of a stream operator.

File: akka-docs/src/main/paradox/stream/operators/Source-or-Flow/filter.md [hunk 1]
```diff
@@ -29,3 +29,6 @@ it returns false the element is discarded.
 
 @@@
 
+## API docs
+
+@apidoc[Flow.filter](Flow) { scala="#filter(p:Out=%3EBoolean):FlowOps.this.Repr[Out]" java="#filter(akka.japi.function.Predicate)" }
```
Comment (raboof) [inline akka-docs/src/main/paradox/stream/operators/Source-or-Flow/filter.md]: The java version doesn't seem to work for me: it links to https://doc.akka.io/japi/akka/2.6/?akka/stream/javadsl/Flow.html#filter(akka.japi.function.Predicate) , and the inner summary correctly switches to the `Flow` page but not to that id in the `Flow` page... is there something we can do about that?

I guess if not then it's still OK to link to the right method for scaladoc and to the class for javadoc

Comment (ennru) [reply akka-docs/src/main/paradox/stream/operators/Source-or-Flow/filter.md]: The frames break it. The anchor is not picked up by the Javadoc page.
We could use direct links without the `?`.
Should we even switch to Javadoc 11 to get rid of the frames altogether?

Comment (raboof) [reply akka-docs/src/main/paradox/stream/operators/Source-or-Flow/filter.md]: actually I think we're already using jdk11 to generate the javadocs, but still using `--frames`. I guess we can/should remove that as it is 'the future' anyway...

**Comment by ennru** [comment]
I removed the `--frames` flag.

**Review by raboof — APPROVED** [review]
Checked links now indeed work

**Pull request merged**